modin/pandas/dataframe.py

```diff
@@ -795,25 +795,40 @@ def equals(self, other):  # noqa: PR01, RT01, D200
             and self.eq(other).all().all()
         )
 
-    def _update_var_dicts_in_kwargs(self, expr, kwargs):
+    def _update_var_dicts_in_kwargs(self, expr, level: int, kwargs):
         """
         Copy variables with "@" prefix in `local_dict` and `global_dict` keys of kwargs.
 
+        This function exists because we neeed to infer local and variables
+        ourselves here in the main node instead of having the remote functions
+        infer them on their own. The remote functions take place in a different
+        thread with their own stacks that normally do not match the stack that
+        leads to the Modin eval or query call.
+
         Parameters
         ----------
         expr : str
             The expression string to search variables with "@" prefix.
+        level : int
+            The number of prior stack frames back to look for local variables.
+            level=0 means look just one frame back.
         kwargs : dict
             See the documentation for eval() for complete details on the keyword arguments accepted by query().
         """
         if "@" not in expr:
             return
-        frame = sys._getframe()
-        try:
-            f_locals = frame.f_back.f_back.f_back.f_back.f_locals
-            f_globals = frame.f_back.f_back.f_back.f_back.f_globals
-        finally:
-            del frame
+
+        # Bump level up one because this function is wrapped in the Modin
+        # logging function wrapper. The alternative is for the wrapper to
+        # give the Modin functions that deal with `level` the special
+        # treatment of bumping `level` up by one, but that's not so nice
+        # either.
+        level += 1
```

Comment:
I wonder if we should just discard any `modin`-owned frames (or at least `logging`-owned) instead of calculating this offset manually